src/network/networkd-link.c

```diff
@@ -152,6 +152,42 @@ static bool link_ipv6ll_enabled(Link *link) {
         return link->network->link_local & ADDRESS_FAMILY_IPV6;
 }
 
+bool link_ipv4ll_fallback_enabled(Link *link) {
```

Comment:
This is almost equivalent to `link_ipv4ll_enabled()` except for the last line. So, how about to add an extra argument `bool allow_fallback` for `link_ipv4ll_enabled()` instead to add this function?

Reply:
yes but other such functions are also very similar. I think this is a clear way of doing rather breaking a pattern we using.

Reply:
OK. Let's leave the cleanups for later PRs.